spec/models/procedure_spec.rb

```diff
@@ -536,38 +536,77 @@
 
     after { Timecop.return }
 
-    context "without parent procedure" do
+    context "without canonical procedure" do
       before do
         Timecop.freeze(now)
         procedure.publish!
       end
 
       it do
+        expect(procedure.canonical_procedure).to be_nil
         expect(procedure.closed_at).to be_nil
         expect(procedure.published_at).to eq(now)
         expect(Procedure.find_by(path: "example-path")).to eq(procedure)
         expect(Procedure.find_by(path: "example-path").administrateurs).to eq(procedure.administrateurs)
       end
     end
+
+    context "with canonical procedure" do
+      let(:canonical_procedure) { create(:procedure, :published) }
+
+      before do
+        Timecop.freeze(now)
+        procedure.publish!(canonical_procedure)
+      end
+
+      it do
+        expect(procedure.canonical_procedure).to eq(canonical_procedure)
+        expect(procedure.closed_at).to be_nil
+        expect(procedure.published_at).to eq(now)
+      end
+    end
   end
 
   describe "#publish_or_reopen!" do
-    let(:published_procedure) { create(:procedure, :published) }
-    let(:administrateur) { published_procedure.administrateurs.first }
+    let(:canonical_procedure) { create(:procedure, :published) }
+    let(:administrateur) { canonical_procedure.administrateurs.first }
 
     let(:procedure) { create(:procedure, administrateurs: [administrateur]) }
     let(:now) { Time.zone.now.beginning_of_minute }
 
-    context "without parent procedure" do
+    context "without canonical procedure" do
+      before do
+        Timecop.freeze(now)
+        procedure.path = canonical_procedure.path
+        procedure.publish_or_reopen!(administrateur)
+        canonical_procedure.reload
+      end
+
+      it do
```

Comment:
Pour moi une bonne partie de la valeur des specs, c'est d'expliquer aux humains le comportement attendu.

Du coup : 

- Je mettrais bien une description aux blocs `it '…'`
- Je séparerais bien le comportement de base ("à la fin la démarche est publiée") des détails ("et voici le cas où on a une démarche canonique").

Un truc du genre :

```ruby
    it 'changes the procedure state to published' do
      expect(procedure.published_at).to eq(now)
      expect(procedure.closed_at).to be_nil
    end

    context 'when publishing over a previous canonical procedure' do
      before do
        # …
      end

      it 'unpublishes the canonical procedure' do
        expect(parent_procedure.unpublished_at).to eq(now)
      end

      it 'references the canonical procedure on the published procedure' do
        expect(procedure.canonical_procedure).to eq(canonical_procedure)
      end 
    end
```

Mais bon, c'est p-ê du pinaillage :)

Reply:
done